qiskit_nature/circuit/library/ansatzes/succ_full.py

```diff
@@ -0,0 +1,208 @@
+        alpha_excitations = get_alpha_excitations(
+            num_electrons, num_spin_orbitals, self._generalized
+        )
+        beta_excitations = get_beta_excitations(num_electrons, num_spin_orbitals, self._generalized)
+        logger.debug("Generated list of single alpha excitations: %s", alpha_excitations)
+        logger.debug("Generated list of single beta excitations: %s", beta_excitations)
+
+        # Find all possible double excitations constructed from the list of single excitations.
+        # Note, that we use `product` here, in order to also get those double
+        # excitations form an alpha excitation and a beta excitation. We will need those in the
+        # following post-processing step.
+        pool = itertools.product(alpha_excitations, beta_excitations)
+
+        for exc in pool:
+            # find the two excitations (Note: SUCCD only works for double excitations!)
+            alpha_exc, second_exc = exc[0], exc[1]
+            # shift the second excitation into the beta-spin orbital index range
+            beta_exc = (
+                second_exc[0],
+                second_exc[1],
+            )
+            # add the excitation tuple
+            occ: Tuple[int, ...]
+            unocc: Tuple[int, ...]
+            occ, unocc = zip(alpha_exc, beta_exc)
+            exc_tuple = (occ, unocc)
+            excitations.append(exc_tuple)
+            logger.debug("Added the excitation: %s", exc_tuple)
+
+        return excitations
+
+    def _validate_num_particles(self, num_particles):
+        try:
+            assert num_particles[0] == num_particles[1]
+        except AssertionError as exc:
+            raise QiskitNatureError(
+                "The SUCC_full Ansatz only works for singlet-spin systems. However, you specified "
+                "differing numbers of alpha and beta electrons:",
+                str(num_particles),
+            ) from exc
+
+    def _build_fermionic_excitation_ops(self, excitations: Sequence) -> List[FermionicOp]:
+        """Builds all possible excitation operators with the given number of excitations for the
+        specified number of particles distributed in the number of orbitals.
+        Args:
+            excitations: the list of excitations.
+        Returns:
+            The list of excitation operators in the second quantized formalism.
+        """
+        operators = []
+        excitations_dictionary: Dict[int, List[Tuple[Tuple[int, ...], Tuple[int, ...]]]] = {}
+        for exc in excitations:
+            exc_level = sum(exc[1])
+            if exc_level in excitations_dictionary:
+                excitations_dictionary[exc_level].append(exc)
+            else:
+                excitations_dictionary[exc_level] = [exc]
+        for exc_level, exc_level_items in excitations_dictionary.items():
+            final_op = FermionicOp.zero(register_length=self.num_spin_orbitals)
```

Comment:
The way you initialize this, will actually result in an additional term which I would like to avoid. Maybe you can gather the individual ops in a list instead, which you then sum over?
E.g.
```python
ops = []
for exc in exc_level_items:
    # ...
    ops.append(op)
operators.append(sum(ops))
```

Reply:
I have problem with this part: To avoid the additional term, I used the code you showed here. When I run "make mypy" , I got an error: **Incompatible return value type (got "List[Union[FermionicOp, int]]", expected "List[FermionicOp]")**. I have no idea where the int come from.

Reply:
In line 233 you have this `operators = []` which has no type. Then in the return, after appending things to this, you have different return paths. If not operators path is clearly a List of FermionicOps. The other path is operators which is this Literal. So in total it thinks the return is a Union of Fermionic and Literal list - but the return type hint on the method definition says FermionicOp list. If you change line 233 to `operators: List[FermionicOp] = []` so you give it the type upfront then I think that will work.


You also have another error related to `pool`. mypy picks up on the type of a local variable from its first use. When you use the same variable below it complains because the type being assigned does not match what the type it thinks the var is. Easiest fix if to call the second pool some other name e.g pool_full in the second case and do for exc in pool_full there. Honestly I am not sure why we even need the local var pool as it used only once so `for exc in itertools.product(....` in each case would get rid of it completely and you would not have the problem with how a local var is used.

Reply:
Thanks for the solution! Now `operators` is fine, but I face the same problem on `ops`. No matter I use `ops = []` or , `ops: List[FermionicOp] = []`, it warns **Argument 1 to "append" of "list" has incompatible type "int"; expected "FermionicOp"  [arg-type]**

Reply:
That's from line 264 i.e. `operators.append(sum(ops))` according to CI. `operators` is a List[FermionicOp] and as you indicate so is ops. But I imagine `sum(ops)` perhaps mypy imagines is summing to an int whereas this adds FermionicOps together right for a single FermionicOp. Maybe use the typing cast construct to inform mypy that the type from sum(ops) is a FermionicOp and not an int. At least this is my best guess!